### PR TITLE
Add-704_binary_search.

### PR DESCRIPTION
**Pull request by Amoonguses1**
binary search

### CODE REVIEW

File: 704_binary_search.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# time: O(n) n is nums length
```
Comment (tkkawa) [inline 704_binary_search.py:1]: O(logn)

File: 704_binary_search.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+from typing import List
+
+
+class Solution:
+
+    def search(self, nums: List[int], target: int) -> int:
+        left = -1
+        right = len(nums)
+        while left < right - 1:
+            now = (left + right)//2
```
Comment (tkkawa) [inline 704_binary_search.py:12]: See #22 .

**Review by tkkawa — CHANGES_REQUESTED** [review]

**Comment by Amoonguses1** [comment]
I fixed the parts reviewer pointed out. I added error handling and docstring. Please check my code.
@tkkawa

File: 704_binary_search.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        if not nums:
+            raise ValueError("nums should not be empty.")
+
+        if target is None:
+            raise ValueError("target should not be empty.")
+
+        left = -1
+        right = len(nums)
+        while left < right - 1:
+            now = (left+right)//2
```
Comment (tkkawa) [inline 704_binary_search.py:28]: Put whitespace before and after "//"

**Review by tkkawa — CHANGES_REQUESTED** [review]

**Comment by Amoonguses1** [comment]
I have fixed to follow coding style.
@tkkawa

**Review by tkkawa — APPROVED** [review]
LGTM

**Pull request merged**